<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.551">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>ushio-ecolab - Publications</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">ushio-ecolab</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./research.html"> 
<span class="menu-text">Research</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./member.html"> 
<span class="menu-text">Members</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./publication.html" aria-current="page"> 
<span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./joinlab.html"> 
<span class="menu-text">How to join the lab</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./japanese.html"> 
<span class="menu-text">Japanese (日本語)</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#representative-papers" id="toc-representative-papers" class="nav-link active" data-scroll-target="#representative-papers"><font color="darkblue">Representative papers</font></a></li>
  <li><a href="#published-preprints" id="toc-published-preprints" class="nav-link" data-scroll-target="#published-preprints">Published / Preprints</a>
  <ul class="collapse">
  <li><a href="#preprints" id="toc-preprints" class="nav-link" data-scroll-target="#preprints"><font color="gray">Preprints</font></a></li>
  <li><a href="#in-press-early-view" id="toc-in-press-early-view" class="nav-link" data-scroll-target="#in-press-early-view">In press / early view</a></li>
  <li><a href="#section" id="toc-section" class="nav-link" data-scroll-target="#section">2025</a></li>
  <li><a href="#section-1" id="toc-section-1" class="nav-link" data-scroll-target="#section-1">2024</a></li>
  <li><a href="#section-2" id="toc-section-2" class="nav-link" data-scroll-target="#section-2">2023</a></li>
  <li><a href="#before-joining-hkust" id="toc-before-joining-hkust" class="nav-link" data-scroll-target="#before-joining-hkust"><font color="gray">Before joining HKUST</font></a></li>
  <li><a href="#section-3" id="toc-section-3" class="nav-link" data-scroll-target="#section-3">2022</a></li>
  <li><a href="#section-4" id="toc-section-4" class="nav-link" data-scroll-target="#section-4">2021</a></li>
  <li><a href="#section-5" id="toc-section-5" class="nav-link" data-scroll-target="#section-5">2020</a></li>
  <li><a href="#section-6" id="toc-section-6" class="nav-link" data-scroll-target="#section-6">2019</a></li>
  <li><a href="#section-7" id="toc-section-7" class="nav-link" data-scroll-target="#section-7">2018</a></li>
  <li><a href="#section-8" id="toc-section-8" class="nav-link" data-scroll-target="#section-8">2017</a></li>
  <li><a href="#section-9" id="toc-section-9" class="nav-link" data-scroll-target="#section-9">2016</a></li>
  <li><a href="#section-10" id="toc-section-10" class="nav-link" data-scroll-target="#section-10">2015</a></li>
  <li><a href="#section-11" id="toc-section-11" class="nav-link" data-scroll-target="#section-11">2014</a></li>
  <li><a href="#section-12" id="toc-section-12" class="nav-link" data-scroll-target="#section-12">2013</a></li>
  <li><a href="#before-2012" id="toc-before-2012" class="nav-link" data-scroll-target="#before-2012">Before 2012</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Publications</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>See <a href="https://scholar.google.com/citations?user=GtIzgfYAAAAJ" target="_blank">Google Scholar</a> for the citation information.</p>
<section id="representative-papers" class="level1">
<h1><font color="darkblue">Representative papers</font></h1>
<ul>
<li><strong>Ushio M</strong>*, Saito H, Tojo M, Nagano AJ (2023) An ecological network approach for detecting and validating influential organisms for rice growth. <a href="https://doi.org/10.7554/eLife.87202"><em>eLife</em> 12:RP87202</a></li>
<li><strong>Ushio M</strong>*, Sado T, Fukuchi T, Sasano S, Masuda R, Osada Y, Miya M* (2023) Temperature sensitivity of the interspecific interaction strength of coastal marine fish communities. <a href="https://doi.org/10.7554/eLife.85795"><em>eLife</em> 12:RP85795</a></li>
<li><strong>Ushio M</strong>*, Watanabe K, Fukuda Y, Tokudome Y, Nakajima K* (2023) Computational capability of ecological dynamics. <a href="https://doi.org/10.1098/rsos.221614"><em>Royal Society Open Science</em> 10:221614</a></li>
<li><strong>Ushio M</strong>* (2022) Interaction capacity as a potential driver of community diversity. <a href="https://doi.org/10.1098/rspb.2021.2690"><em>Proceedings of the Royal Society B</em> 289: 20212690</a> <!--
-  **Ushio M**\* (2019) Use of a filter cartridge combined with intra-cartridge bead-beating improves detection of microbial DNA from water samples. [_Methods in Ecology and Evolution_ 10 (8):1142–1156](https://doi.org/10.1111/2041-210X.13204)

-  **Ushio M**\*, Murakami H, Masuda R, Sado T, Miya M, Sakurai S, Yamanaka H, Minamoto T, Kondoh M (2018) Quantitative monitoring of multispecies fish environmental DNA using high-throughput sequencing.  [_Metabarcoding & Metagenomics_ 2: e23297](https://dx.doi.org/10.3897/mbmg.2.23297)
--></li>
<li><strong>Ushio M</strong>*, Hsieh CH, Masuda R, Deyle RE, Ye H, Chang CW, Sugihara G, Kondoh M* (2018) Fluctuating interaction network and time-varying stability of a natural fish community. <a href="https://doi.org/10.1038/nature25504"><em>Nature</em> 554 (7692): 360–363</a></li>
<li><strong>Ushio M</strong>*, Fukuda H, Inoue T, Makoto K, Kishida O, Sato K, Murata K, Nikaido M, Sado T, Sato Y, Takeshita M, Iwasaki W, Yamanaka H, Kondoh M, Miya M* (2017) Environmental DNA enables detection of terrestrial mammals from forest pond water. <a href="https://dx.doi.org/10.1111/1755-0998.12690"><em>Molecular Ecology Resources</em> 17: e63–e75</a></li>
<li><strong>Ushio M</strong>*, Aiba S-I, Takeuchi Y, Iida Y, Matsuoka S, Repin R, Kitayama K (2017) Plant-soil feedbacks and the dominance of conifers in a tropical montane forest in Borneo. <a href="https://dx.doi.org/10.1002/ecm.1236"><em>Ecological Monographs</em> 87 (1): 105–129</a></li>
</ul>
<p><br></p>
</section>
<section id="published-preprints" class="level1">
<h1>Published / Preprints</h1>
<hr>
<section id="preprints" class="level3">
<h3 class="anchored" data-anchor-id="preprints"><font color="gray">Preprints</font></h3>
<hr>
<p><font color="gray"></font></p><font color="gray">
<ul>
<li><p>Gao et al.&nbsp;(2024) Diversity and distinctive traits of the global RNA virome in urban environments. <a href="https://dx.doi.org/10.2139/ssrn.4871972">Available at SSRN</a></p></li>
<li><p>Osada Y*, <strong>Ushio M</strong>, Kondoh M (2023) A unified framework for nonparametric causality detection. <a href="https://doi.org/10.1101/2023.04.20.537743"><em>bioRxiv</em></a></p></li>
</ul>
</font><p><font color="gray"></font></p>
</section>
<section id="in-press-early-view" class="level2">
<h2 class="anchored" data-anchor-id="in-press-early-view">In press / early view</h2>
<ol type="1">
<li><strong>Ushio M</strong>*, Nakajima K (in press)<br>
Reservoir computing: theory and its potential applications in ecology. (in Japanese) <em>Japanese Journal of Ecology</em></li>
</ol>
</section>
<section id="section" class="level2">
<h2 class="anchored" data-anchor-id="section">2025</h2>
<ol type="1">
<li><p>Miki T*, Chang CW, Ke PJ, Telschow A, Tsai CH, <strong>Ushio M</strong>, Hsieh CH (2025) How to quantify interaction strengths? A critical rethinking of the interaction Jacobian and evaluation methods for non-parametric inference in time series analysis. <a href="https://doi.org/10.1016/j.physd.2025.134613"><em>Physica D: Nonlinear Phenomena</em> 134613</a></p></li>
<li><p>Fukasawa Y*, Kimura S, Kominami Y, Takagi M, Matsukura K, Makoto K, Suzuki SN, Takemoto S, Tanaka N, Jomura M, Kadowaki K, <strong>Ushio M</strong>, Kinuura H, Yamashita S (2025)<br>
Oak wilt disease may reduce the initial decay rate of dead Quercus serrata stems by altering fungal communities in the wood. <a href="https://doi.org/10.1111/1462-2920.70026"><em>Environmental Microbiology</em> 27:e70026</a></p></li>
</ol>
</section>
<section id="section-1" class="level2">
<h2 class="anchored" data-anchor-id="section-1">2024</h2>
<ol type="1">
<li><p>Oka S-I*, Sado T, <strong>Ushio M</strong>, Miya M* (2024)<br>
Assessing the impact of tidal changes on fish environmental DNA metabarcoding in temperate and tropical coastal regions of Japan. <a href="https://doi.org/10.3897/mbmg.8.135461"><em>Metabarcoding &amp; Metagenomics</em> 8:e135461</a></p></li>
<li><p>Jiang L*, <strong>Ushio M</strong>, Imai N, Sugau J, Kitayama K (2024) Changes of bacterial versus fungal community composition along a forest degradation gradient of logged-over tropical rain forests, and their consequences on soil enzyme activities in Malaysian Borneo. <a href="https://doi.org/10.1007/s11104-024-06939-6"><em>Plant and Soil</em></a></p></li>
<li><p>Atsumi K, Nishida Y, <strong>Ushio M</strong>, Nishi H, Genroku T, Fujiki S* (2024) Boosting biodiversity monitoring using smartphone-driven, rapidly accumulating citizen data. <a href="https://doi.org/10.7554/eLife.93694"><em>eLife</em> 13:RP93694</a></p></li>
<li><p>Maehara K, Iwachido Y, Katsuhara H, Tomitaka M, Seto K, <strong>Ushio M</strong>, Kagami M, Sasaki T* (2024)<br>
Plant and soil microbial community assembly processes across urban vacant lots. <a href="https://doi.org/10.1111/jvs.13262"><em>Journal of Vegetation Science</em> 35:e13262</a></p></li>
<li><p>Sugiyama Y*, Matsuoka S, Shimono Y, <strong>Ushio M</strong>, Doi H (2024)<br>
Do water fungal environmental DNA assemblages reflect the surrounding terrestrial sporocarp communities? <a href="https://doi.org/10.1016/j.funeco.2023.101311"><em>Fungal Ecology</em> 67:101311</a></p></li>
</ol>
</section>
<section id="section-2" class="level2">
<h2 class="anchored" data-anchor-id="section-2">2023</h2>
<ol type="1">
<li>Fujita H*, <strong>Ushio M</strong>, Suzuki K, Abe MS, Yamamichi M, Okazaki Y, Canarini A, Hayashi I, Fukushima K, Fukuda S, Kiers ET, Toju H* (2023) Metagenomic analysis of ecological niche overlap and community collapse in microbiome dynamics. <a href="https://doi.org/10.3389/fmicb.2023.1261137"><em>Frontiers in Microbiology</em> 14:1261137</a></li>
<li>Sakata A*, Sado T, Oka S-I, <strong>Ushio M</strong>, Miya M* (2023)<br>
Collection of environmental DNA from stemflow for monitoring arboreal biodiversity: preliminary validation using lichens. <a href="https://doi.org/10.1016/j.mex.2023.102448"><em>MethodsX</em> 11:102448</a></li>
<li>Maxime M*, <strong>Ushio M</strong>, Sakai S. (2023)<br>
The effects of the floral infection by a bacterial pathogen in a dioecious plant, <em>Mallotus japonicus</em> (Euphorbiaceae). <a href="https://doi.org/10.1002/1438-390X.12110"><em>Population Ecology</em> 65:232-239</a></li>
<li><strong>Ushio M</strong>*, Saito H, Tojo M, Nagano AJ (2023)<br>
An ecological network approach for detecting and validating influential organisms for rice growth. <a href="https://doi.org/10.7554/eLife.87202"><em>eLife</em> 12:RP87202</a></li>
<li><strong>Ushio M</strong>*, Sado T, Fukuchi T, Sasano S, Masuda R, Osada Y, Miya M* (2023)<br>
Temperature sensitivity of the interspecific interaction strength of coastal marine fish communities. <a href="https://doi.org/10.7554/eLife.85795"><em>eLife</em> 12:RP85795</a></li>
<li>Yoneya K*, <strong>Ushio M</strong>, Miki T (2023)<br>
Non-destructive collection and metabarcoding of arthropod environmental DNA remained on a terrestrial plant. <a href="https://doi.org/10.1038/s41598-023-32862-4"><em>Scientific Reports</em> 13:7125</a></li>
<li>Jiang L*, <strong>Ushio M</strong>, Kitayama K (2023)<br>
Changes of soil chemical properties, microbial biomass and enzymatic activities along a gradient of forest degradation in logged over tropical rain forests, Borneo. <a href="https://doi.org/10.1007/s11104-022-05848-w"><em>Plant and Soil</em> 485:525–536</a></li>
<li><strong>Ushio M</strong>*, Watanabe K, Fukuda Y, Tokudome Y, Nakajima K* (2023)<br>
Computational capability of ecological dynamics. <a href="https://doi.org/10.1098/rsos.221614"><em>Royal Society Open Science</em> 10:221614</a></li>
<li>Fujita H*, <strong>Ushio M</strong>, Suzuki K, Abe MS, Yamamichi M, Okazaki Y, Canarini A, Hayashi I, Fukushima K, Fukuda S, Kiers ET, Toju H* (2023)<br>
Facilitative interaction networks in experimental microbial community dynamics. <a href="https://doi.org/10.3389/fmicb.2023.1153952"><em>Frontiers in Microbiology</em> 14:1153952</a></li>
<li>Fujita H*, <strong>Ushio M</strong>, Suzuki K, Abe MS, Yamamichi M, Iwayama K, Canarini A, Hayashi I, Fukushima K, Fukuda S, Kiers ET, Toju H* (2023)<br>
Alternative stable states, nonlinear behavior, and predictability of microbiome dynamics. <a href="https://doi.org/10.1186/s40168-023-01474-5"><em>Microbiome</em> 11:63</a></li>
<li>Fukasawa Y*, Akai D, <strong>Ushio M</strong>, Takehi T (2023)<br>
Electrical potentials in the ectomycorrhizal fungus Laccaria bicolor after a rainfall event. <a href="https://doi.org/10.1016/j.funeco.2023.101229"><em>Fungal Ecology</em> 63:101229</a></li>
<li>Tsujii Y*, Sakai S, <strong>Ushio M</strong>, Aiba S-I, Kitayama K (2023)<br>
Variations in the reproductive cycle of Bornean montane tree species along elevational gradients on ultrabasic and non-ultrabasic soils. <a href="https://doi.org/10.1111/btp.13177"><em>Biotropica</em> 55 (1):210–220</a></li>
</ol>
</section>
<section id="before-joining-hkust" class="level2">
<h2 class="anchored" data-anchor-id="before-joining-hkust"><font color="gray">Before joining HKUST</font></h2>
</section>
<section id="section-3" class="level2">
<h2 class="anchored" data-anchor-id="section-3">2022</h2>
<ol type="1">
<li><strong>Ushio M</strong>*, Furukawa S, Murakami H, Masuda R, Nagano AJ* (2022)<br>
An efficient early-pooling protocol for environmental DNA metabarcoding. <a href="https://doi.org/10.1002/edn3.337"><em>Environmental DNA</em> 4 (6):1212–1228</a></li>
<li>Inoue N*, Sato M, Furuichi N, Imaizumi T, <strong>Ushio M</strong> (2022)<br>
The relationship between eDNA density distribution and current fields around an artificial reef in the waters of Tateyama Bay, Japan. <a href="https://doi.org/10.3897/mbmg.6.87415"><em>Metabarcoding &amp; Metagenomics</em> 6:e87415</a></li>
<li>Fukasawa Y*, Matsukura K, Stephan JG, Makoto K, Suzuki SN, Kominami Y, Takagi M, Tanaka N, Takemoto S, Kinuura H, Okano K, Song Z, Jomura M, Kadowaki K, Yamashita S, <strong>Ushio M</strong> (2022)<br>
Patterns of community composition and diversity in latent fungi of living <em>Quercus serrata</em> trunks across a range of oak wilt prevalence and climate variables in Japan. <a href="https://doi.org/10.1016/j.funeco.2021.101095"><em>Fungal Ecology</em> 59:101095</a></li>
<li>Elvira NJ, <strong>Ushio M</strong>, Sakai S* (2022)<br>
Are microbes growing on flowers evil? Effects of old flower microbes on fruit set in a wild ginger with one-day flowers, <em>Alpinia japonica</em> (Zingiberaceae). <a href="https://doi.org/10.3897/mbmg.6.84331"><em>Metabarcoding &amp; Metagenomics</em> 6:e84331</a></li>
<li>Maxime M*, <strong>Ushio M</strong>, Sakai S. (2022)<br>
Contrasting microbial communities on male and female flowers of a dioecious plant, <em>Mallotus japonicus</em> (Euphorbiaceae). <a href="https://doi.org/10.1002/edn3.271"><em>Environmental DNA</em> 4 (3):565–579</a></li>
<li><strong>Ushio M</strong>* (2022) Interaction capacity as a potential driver of community diversity. <a href="https://doi.org/10.1098/rspb.2021.2690"><em>Proceedings of the Royal Society B</em> 289: 20212690</a></li>
</ol>
</section>
<section id="section-4" class="level2">
<h2 class="anchored" data-anchor-id="section-4">2021</h2>
<ol type="1">
<li>Chang CW, Miki T, <strong>Ushio M</strong>, Ke PJ, Lu HP, Shiah FK, Hsieh CH* (2021)<br>
Reconstructing large interaction networks from empirical time series data. <a href="https://doi.org/10.1111/ele.13897"><em>Ecology Letters</em> 24 (12):2763–2774</a></li>
<li>Sato M*, Inoue N, Nambu R, Furuich N, Imaizumi T, <strong>Ushio M</strong> (2021)<br>
Quantitative assessment of multiple fish species around artificial reefs using environmental DNA metabarcoding. <a href="https://doi.org/10.1038/s41598-021-98926-5"><em>Scientific Reports</em> 11:19477</a></li>
<li>Matsuoka S*, Sugiyama Y, Shimono Y, <strong>Ushio M</strong>, Doi H (2021)<br>
Evaluation of seasonal dynamics of fungal DNA assemblages in a flow-regulated stream in a restored forest using eDNA metabarcoding. <a href="https://doi.org/10.1111/1462-2920.15669"><em>Environmental Microbiology</em> 23 (8):4797–4806</a></li>
<li>Doi H*, Yasuhara M*, <strong>Ushio M</strong>* (2021)<br>
Causal analysis of the temperature impact on deep-sea biodiversity. <a href="https://doi.org/10.1098/rsbl.2020.0666"><em>Biology Letters</em> 17 (7):20200666</a></li>
<li>Shiragaki T, Inoue T, Fukuda H, <strong>Ushio M</strong>, Kusakabe M, Okano T, Takasu H* (2021)<br>
A pilot study investigating fish species composition using environmental DNA in marine sediment (in Japanese). <a href="https://doi.org/10.2965/jswe.44.79"><em>Journal of Japan Society on Water Environment</em> 44 (4):79–84</a></li>
<li>Kawatsu K, <strong>Ushio M</strong>, van Veen F, Kondoh M* (2021)<br>
Are networks of trophic interactions sufficient for understanding the dynamics of multi-trophic communities? Analysis of a tri-trophic insect food-web time-series. <a href="https://doi.org/10.1111/ele.13672"><em>Ecology Letters</em> 24 (3):543–552</a></li>
<li>Kitayama K*, <strong>Ushio M</strong>*, Aiba S-I (2021)<br>
Temperature is a dominant driver of distinct annual seasonality of leaf litter production of equatorial tropical rain forests. <a href="https://doi.org/10.1111/1365-2745.13500"><em>Journal of Ecology</em> 109 (2):727–736</a></li>
</ol>
</section>
<section id="section-5" class="level2">
<h2 class="anchored" data-anchor-id="section-5">2020</h2>
<ol type="1">
<li>Yonezawa S, <strong>Ushio M</strong>*, Yamanaka H, Miya M, Takayanagi A, Isagi Y* (2020)<br>
Environmental DNA metabarcoding reveals the presence of a small, quick-moving, nocturnal water shrew in a forest stream. <a href="https://doi.org/10.1007/s10592-020-01310-5"><em>Conservation Genetics</em> 21 (6):1079–1084</a></li>
<li>Tsuji S*, Shibata N, Sawada H, <strong>Ushio M</strong> (2020)<br>
Quantitative evaluation of intraspecific genetic diversity in a natural fish population using environmental DNA analysis. <a href="https://doi.org/10.1111/1755-0998.13200"><em>Molecular Ecology Resources</em> 20 (5):1323–1332</a></li>
<li>Tsuji, S*, Maruyama A, Miya M, <strong>Ushio M</strong>, Sato H, Minamoto T, Yamanaka H (2020)<br>
Environmental DNA analysis shows high potential as a tool for estimating intraspecific genetic diversity in a wild fish population. <a href="https://doi.org/10.1111/1755-0998.13165"><em>Molecular Ecology Resources</em> 20 (5):1248–1258</a></li>
<li>Cai J, Hodoki Y, <strong>Ushio M</strong>, Nakano S-I* (2020)<br>
Influences of potential grazers on picocyanobacteria abundances in Lake Biwa revealed by empirical dynamic modeling. <a href="https://doi.org/10.1080/20442041.2020.1711682"><em>Inland Water</em> 10 (3):386–396</a></li>
<li><strong>Ushio M</strong>* (2020) [Special Feature]<br>
Idea paper: Predicting culturability of microbes from population dynamics under field conditions. <a href="https://doi.org/10.1111/1440-1703.12104"><em>Ecological Research</em> 35 (4):586–590</a></li>
<li>Nishio H*, Buzas DM, Nagano AJ, Iwayama K, <strong>Ushio M</strong>, Kudoh H* (2020)<br>
Repressive chromatin modification underpins the long-term expression trend of a perennial flowering gene in nature. <a href="https://doi.org/10.1038/s41467-020-15896-4"><em>Nature Communications</em> 11:2065</a></li>
<li>Shirouzu T*, Matsuoka S, Doi H, Nagata N, <strong>Ushio M</strong>, Hosaka K (2020) Complementary molecular methods reveal a comprehensive phylogenetic diversity with integrating invisible lineages of early-diverged mushrooms. <a href="https://doi.org/10.1038/s41598-020-59620-0"><em>Scientific Reports</em> 10:3057</a></li>
<li><strong>Ushio M</strong>*, Osada Y, Kumagai T, Kume T, Pungga RS, Nakashizuka T, Itioka T, Sakai S* (2020) Dynamic and synergistic influences of air temperature and rainfall on general flowering in a Bornean lowland tropical forest. <a href="https://doi.org/10.1111/1440-1703.12057"><em>Ecological Research</em> 35 (1):17–29</a></li>
<li>Tsuji S*, Miya M, <strong>Ushio M</strong>, Sato H, Minamoto T, Yamanaka H (2020)<br>
Evaluating intraspecific genetic diversity using environmental DNA and denoising approach: A case study using tank water. <a href="https://doi.org/10.1002/edn3.44"><em>Environmental DNA</em> 2 (1): 42–52</a></li>
</ol>
</section>
<section id="section-6" class="level2">
<h2 class="anchored" data-anchor-id="section-6">2019</h2>
<ol type="1">
<li><strong>Ushio M</strong>* (2019)<br>
Use of a filter cartridge combined with intra-cartridge bead-beating improves detection of microbial DNA from water samples. <a href="https://doi.org/10.1111/2041-210X.13204"><em>Methods in Ecology and Evolution</em> 10 (8):1142–1156</a></li>
<li>Nakagawa M*, <strong>Ushio M</strong>, Kume T, Nakashizuka T (2019)<br>
Seasonal and long-term patterns in litterfall in a Bornean tropical rainforest. <a href="https://doi.org/10.1111/1440-1703.1003"><em>Ecological Research</em> 34 (1): 31–39</a></li>
</ol>
</section>
<section id="section-7" class="level2">
<h2 class="anchored" data-anchor-id="section-7">2018</h2>
<ol type="1">
<li>Toju H*, Peay KG, Yamamichi M, Narisawa K, Hiruma K, Naito K, Fukuda S, <strong>Ushio M</strong>, Nakaoka S, Onoda Y, Yoshida K, Schlaeppi K, Bai Y, Sugiura R, Ichihashi Y, Minamisawa K, Kiers ET (2018)<br>
Core microbiomes for sustainable agroecosystems. <a href="https://dx.doi.org/10.1038/s41477-018-0139-4"><em>Nature Plants</em> 4: 247–257</a></li>
<li><strong>Ushio M</strong>*, Murata K, Sado T, Nishiumi I, Takeshita M, Iwasaki W, Miya M* (2018)<br>
Demonstration of the potential of environmental DNA as a tool for the detection of avian species. <a href="https://dx.doi.org/10.1038/s41598-018-22817-5"><em>Scientific Reports</em> 8: 4493</a></li>
<li><strong>Ushio M</strong>*, Murakami H, Masuda R, Sado T, Miya M, Sakurai S, Yamanaka H, Minamoto T, Kondoh M (2018) Quantitative monitoring of multispecies fish environmental DNA using high-throughput sequencing. <a href="https://dx.doi.org/10.3897/mbmg.2.23297"><em>Metabarcoding &amp; Metagenomics</em> 2: e23297</a></li>
<li><strong>Ushio M</strong>*, Hsieh CH, Masuda R, Deyle RE, Ye H, Chang CW, Sugihara G, Kondoh M* (2018)<br>
Fluctuating interaction network and time-varying stability of a natural fish community. <a href="https://doi.org/10.1038/nature25504"><em>Nature</em> 554 (7692): 360–363</a></li>
</ol>
</section>
<section id="section-8" class="level2">
<h2 class="anchored" data-anchor-id="section-8">2017</h2>
<ol type="1">
<li>Aiba S-I*, Miyamoto K, <strong>Ushio M</strong>, Aoyagi R, Sawada Y (2017)<br>
Mechanism determining conifer dominance in terrestrial ecosystems: A synthesis (in Japanese). <a href="https://doi.org/10.18960/seitai.67.3_355"><em>Japanese Journal of Ecology</em> 67 (3): 355–360</a></li>
<li><strong>Ushio M</strong>* (2017)<br>
Root morphology, mycorrhizal fungi and nitrogen fixation capacity of Podocarpaceae and Araucariaceae (in Japanese). <a href="https://doi.org/10.18960/seitai.67.3_339"><em>Japanese Journal of Ecology</em> 67 (3): 339–345</a></li>
<li><strong>Ushio M</strong>*, Fukuda H, Inoue T, Makoto K, Kishida O, Sato K, Murata K, Nikaido M, Sado T, Sato Y, Takeshita M, Iwasaki W, Yamanaka H, Kondoh M, Miya M* (2017)<br>
Environmental DNA enables detection of terrestrial mammals from forest pond water. <a href="https://dx.doi.org/10.1111/1755-0998.12690"><em>Molecular Ecology Resources</em> 17 (6): e63–e75</a></li>
<li>Chang CW*, <strong>Ushio M</strong>*, Hsieh CH* (2017)<br>
Empirical dynamic modeling for beginners. <a href="https://doi.org/10.1007/s11284-017-1469-9"><em>Ecological Research</em> 32 (6): 785–796</a></li>
<li>Ishige T, Miya M, <strong>Ushio M</strong>, Sado T, Ushioda M, Yonechi R, Lagan P, Matsubayashi H* (2017)<br>
Tropical-forest mammals as detected by environmental DNA at natural saltlicks. <a href="https://doi.org/10.1016/j.biocon.2017.04.023"><em>Biological Conservation</em> 210: 281–285</a></li>
<li>Tsuji S*, <strong>Ushio M</strong>, Sakurai S, Minamoto T, Yamanaka H (2017)<br>
Water temperature-dependent degradation of environmental DNA and its relation to bacterial abundance. <a href="https://doi.org/10.1371/journal.pone.0176608"><em>PLoS ONE</em> 12 (4) e0176608</a></li>
<li><strong>Ushio M</strong>*, Aiba S-I, Takeuchi Y, Iida Y, Matsuoka S, Repin R, Kitayama K (2017)<br>
Plant-soil feedbacks and the dominance of conifers in a tropical montane forest in Borneo. <a href="https://dx.doi.org/10.1002/ecm.1236"><em>Ecological Monographs</em> 87 (1): 105–129</a></li>
<li>Koyama K*, Yamamoto K, <strong>Ushio M</strong> (2017)<br>
A lognormal distribution of the lengths of terminal twigs on self-similar branches of elm trees. <a href="https://dx.doi.org/10.1098/rspb.2016.2395"><em>Proceedings of the Royal Society B</em> 284: 20162395</a></li>
</ol>
</section>
<section id="section-9" class="level2">
<h2 class="anchored" data-anchor-id="section-9">2016</h2>
<ol type="1">
<li>Sakai S*, Keong CY, Koizumi M, Kishimoto-Yamada K, Takano KT, Ichikawa M, Samejima H, Kato Y, Soda R, <strong>Ushio M</strong>, Saizen I, Nakashizuka T, Itioka T (2016)<br>
Social and ecological factors associated with the use of non-timber forest products by people in rural Borneo. <a href="https://dx.doi.org/10.1016/j.biocon.2016.10.022"><em>Biological Conservation</em> 204: 340–349</a></li>
</ol>
</section>
<section id="section-10" class="level2">
<h2 class="anchored" data-anchor-id="section-10">2015</h2>
<ol type="1">
<li>Ohba S-Y*, <strong>Ushio M</strong> (2015)<br>
Effect of water depth on predation frequency by diving beetles on mosquito larvae prey. <a href="https://dx.doi.org/10.1111/ens.12143"><em>Entomological Science</em> 18 (4): 519–522</a></li>
<li><strong>Ushio M</strong>*, Fujiki Y, Hidaka A, Kitayama K* (2015)<br>
Linkage of root physiology and morphology as an adaptation to soil phosphorus impoverishment in tropical montane forests. <a href="https://dx.doi.org/10.1111/1365-2435.12424"><em>Functional Ecology</em> 29 (9): 1235–1245</a></li>
<li>Takasu H, <strong>Ushio M</strong>, LeClair JE, Nakano S-I* (2015)<br>
High contribution of <em>Synechococcus</em> to phytoplankton biomass in the aphotic hypolimnion in a deep freshwater lake (Lake Biwa, Japan). <a href="https://dx.doi.org/10.3354/ame01749"><em>Aquatic Microbial Ecology</em> 75 (1): 69–79</a></li>
<li><strong>Ushio M</strong>*, Yamasaki E, Takasu H, Nagano AJ, Honjo MN, Fujinaga S, Ikemoto M, Sakai S, Kudoh H (2015)<br>
Microbial communities on flower surfaces act as signatures of pollinator visitation. <a href="https://dx.doi.org/10.1038/srep08695"><em>Scientific Reports</em> 5: 8695</a></li>
</ol>
</section>
<section id="section-11" class="level2">
<h2 class="anchored" data-anchor-id="section-11">2014</h2>
<ol type="1">
<li>Katayama N*, Silva AO, Kishida O, <strong>Ushio M</strong>, Kita S, Ohgushi T (2014)<br>
Herbivorous insect decreases plant nutrient uptake: the role of soil nutrient availability and association of below-ground symbionts. <a href="https://dx.doi.org/10.1111/een.12125"><em>Ecological Entomology</em> 39 (4): 511–518</a></li>
<li><strong>Ushio M</strong>*, Makoto K, Klaminder J, Takasu H, Nakano S-I (2014)<br>
High-throughput sequencing shows inconsistent results with a microscope-based analysis of the soil prokaryotic community. <a href="https://dx.doi.org/10.1016/j.soilbio.2014.05.010"><em>Soil Biology &amp; Biochemistry</em> 76: 53–56</a></li>
</ol>
</section>
<section id="section-12" class="level2">
<h2 class="anchored" data-anchor-id="section-12">2013</h2>
<ol type="1">
<li><strong>Ushio M</strong>*, Miki T, Balser TC (2013)<br>
A coexisting fungal-bacterial community stabilizes soil decomposition activity in a microcosm experiment. <a href="https://dx.plos.org/10.1371/journal.pone.0080320"><em>PLoS ONE</em> 8 (11) e80320</a></li>
<li><strong>Ushio M</strong>*, Makoto K, Klaminder J, Nakano S-I (2013)<br>
CARD-FISH analysis of prokaryotic community composition and abundance along small-scale vegetation gradients in a dry arctic tundra ecosystem. <a href="https://dx.doi.org/10.1016/j.soilbio.2013.05.002"><em>Soil Biology &amp; Biochemistry</em> 64: 147–154</a></li>
<li>Nakazawa T*, Ohba S-Y, <strong>Ushio M</strong> (2013)<br>
Predator-prey body size relationships when predators can consume prey larger than themselves. <a href="https://dx.doi.org/10.1098/rsbl.2012.1193"><em>Biology Letters</em> 9 (3) no. 20121193</a></li>
<li><strong>Ushio M</strong>*, Balser TC, Kitayama K (2013)<br>
Effects of condensed tannins in conifer leaves on the composition and activity of soil microbial community in a tropical montane forest. <a href="https://dx.doi.org/10.1007/s11104-012-1365-6"><em>Plant and Soil</em> 365 (1–2): 157–170</a></li>
</ol>
</section>
<section id="before-2012" class="level2">
<h2 class="anchored" data-anchor-id="before-2012">Before 2012</h2>
<ol type="1">
<li>Koyama K*, Hidaka Y, <strong>Ushio M</strong> (2012)<br>
Dynamic scaling in the growth of a non-branching plant, Cardiocrinum cordatum. <a href="https://dx.plos.org/10.1371/journal.pone.0045317"><em>PLoS ONE</em> 7 (9) e45317</a></li>
<li><strong>Ushio M</strong>, Adams JM* (2011) A meta-analysis of the global distribution pattern of condensed tannins in tree leaves. <a href="http://benthamopen.com/ABSTRACT/TOECOLJ-4-18"><em>The Open Ecology Journal</em> 4: 18–23</a><br>
</li>
<li>Takeuchi Y, Salcher MM, <strong>Ushio M</strong>, Shimizu-Inatsugi R, Kobayashi MJ, Diway B, von Mering C, Pernthaler J, Shimizu KK* (2011)<br>
<em>In situ</em> enzyme activity in the dissolved and particulate fraction of the fluid from four pitcher plant species of the genus <em>Nepenthes</em>. <a href="https://dx.doi.org/10.1371/journal.pone.0025144"><em>PLoS ONE</em> 6 (9) e25144</a><br>
</li>
<li><strong>Ushio M</strong>*, Kitayama K, Balser TC (2010)<br>
Tree species-mediated spatial patchiness of the composition of microbial community and physicochemical properties in the topsoils of a tropical montane forest. <a href="https://dx.doi.org/10.1016/j.soilbio.2010.05.035"><em>Soil Biology &amp; Biochemistry</em> 42 (9): 1588–1595</a><br>
</li>
<li>Miki T*, <strong>Ushio M</strong>, Fukui S, Kondoh M (2010)<br>
Functional diversity of microbial decomposers facilitates plant coexistence in a plant-microbe-soil feedback model. <a href="https://dx.doi.org/10.1073/pnas.0914281107"><em>PNAS</em> 107 (32): 14251–14256</a></li>
<li><strong>Ushio M</strong>*, Kitayama K, Balser TC (2010) Tree species effects on soil enzyme activities through effects on soil physicochemical and microbial properties in a tropical montane forest on Mt. Kinabalu, Borneo. <a href="https://dx.doi.org/10.1016/j.pedobi.2009.12.003"><em>Pedobiologia</em> 53 (4): 227–233</a></li>
<li><strong>Ushio M</strong>*, Miki T, Kitayama K (2009) Phenolic control of plant nitrogen acquisition through the inhibition of soil microbial decomposition processes: A plant-microbe competition model. <a href="https://www.jstage.jst.go.jp/article/jsme2/24/2/24_ME09107/_article"><em>Microbes and Environments</em> 24 (2): 180–187</a></li>
<li><strong>Ushio M</strong>*, Wagai R, Balser TC, Kitayama K (2008) Variations in the soil microbial community composition of a tropical montane forest ecosystem: Does tree species matter? <a href="https://dx.doi.org/10.1016/j.soilbio.2008.06.023"><em>Soil Biology &amp; Biochemistry</em> 40 (10): 2699–2702</a></li>
</ol>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>